.github/workflows/CI.yaml

```diff
@@ -0,0 +1,37 @@
+#!————————————————————————————————————————————————————————————————————————————*!
+# GitHub action for continuous integration. Triggered on pull requests into
+# the main branch and commits on the main branch.
+#
+# Currently, this action executes the following:
+#   - Ensure that formatting is correct (no changes when `make clean` is ran)
+#   - Runs unit tests
+#
+#!————————————————————————————————————————————————————————————————————————————*!
+
+name: CI                           # workflow name
+
+on:
+  push:
+    branches: [ $default-branch ]  # Activate on commits to main.
+  pull_request:
+    branches: [ $default-branch ]  # Activate on pull requests.
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2    # Checkout the source code of the project.
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with: |
+        go-version: 1.15
+
+    - name: formatting             # Ensure formatting is correct.
+      run: |
+        make clean
```

Comment:
If we have a pre-commit hook that's already running make clean before any commit, then there's no need to include it again in the workflow.

Reply:
I think the workflow guarantees it. There are a couple of thing it guards against:
  1. Forgetting to set up hooks
  2. Bypassing hooks with `-n`

Reply:
Bro no one's gonna bypass hooks. Anyways, doesn't hurt that much to have it (slows down the CI workflow but not by much) so its fine for now.